skimage/future/_thin_plate_splines.py

```diff
@@ -0,0 +1,323 @@
+        coeffs = self.spline_mappings
+        coords = np.array(coords)
+
+        if not coords.ndim == 2 or coords.shape[1] != 2:
+            raise ValueError("Input `coords` must have shape (N, 2)")
+
+        radial_dist = self._radial_distance(coords[:, 0], coords[:, 1])
+
+        x_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 0]
+        )
+        y_warp = self._spline_function(
+            coords[:, 0], coords[:, 1], radial_dist, coeffs[:, 1]
+        )
+        return np.vstack([x_warp, y_warp]).T
+
+    @property
+    def inverse(self):
+        raise NotImplementedError("This is yet to be implemented.")
+
+    def estimate(self, src, dst):
+        """Estimate optimal spline mappings between source and destination points.
+
+        Parameters
+        ----------
+        src : (N, 2) array_like
+            Control points at source coordinates.
+        dst : (N, 2) array_like
+            Control points at destination coordinates.
+
+        Returns
+        -------
+        success: bool
+            True indicates that the estimation was successful.
+
+        Notes
+        -----
+        The number N of source and destination points must match.
+        """
+
+        check_nD(src, 2)
+        check_nD(dst, 2)
+
+        if len(src) < 3 or len(dst) < 3:
+            raise ValueError(
+                "There should be at least 3 points in both sets (source and destination)."
+            )
+
+        if src.shape != dst.shape:
+            raise ValueError(
+                "The shape of source and destination coordinates must match."
+            )
+
+        self.src = src
+        n, d = src.shape
+
+        dist = sp.spatial.distance.cdist(self.src, self.src)
+        K = _radial_basis_kernel(dist)
+        P = np.hstack([np.ones((n, 1)), src])
+        L = np.zeros((n + 3, n + 3), dtype=np.float32)
+        L[:n, :n] = K
+        L[:n, -3:] = P
+        L[-3:, :n] = P.T
+        V = np.concatenate([dst, np.zeros((d + 1, d))])
+        self.spline_mappings = np.dot(np.linalg.inv(L), V)
+        return True
+
+    def _radial_distance(self, x, y):
+        """Compute the radial distance."""
+        Pi_x = self.src[:, 0]
+        Pi_y = self.src[:, 1]
+
+        dx = Pi_x[:, np.newaxis] - x
+        dy = Pi_y[:, np.newaxis] - y
+        r = np.sqrt(dx**2 + dy**2)
+        radial_dist = _radial_basis_kernel(r)
+        return radial_dist
+
+    def _spline_function(self, x, y, radial_dist, coeffs):
+        """Estimate the spline function in X and Y directions."""
+        w = coeffs[:-3]
+        a1, ax, ay = coeffs[-3:]
+        summation = np.dot(w, radial_dist)
+        return a1 + ax * x + ay * y + summation
+
+
+def _radial_basis_kernel(r):
+    """Compute basis kernel for thin-plate splines.
+
+    Parameters
+    ----------
+    r : (4, N) ndarray
+        Input array representing the norm distance between interlandmark
```

Comment:
What are "norm distance" and "interlandmark"? Consider more common terminology here.